streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -581,8 +580,8 @@ public void punctuate(final ProcessorNode node, final long timestamp, final Punc
         if (processorContext.currentNode() != null) {
             throw new IllegalStateException(format("%sCurrent node is not null", logPrefix));
         }
-
-        updateProcessorContext(new StampedRecord(DUMMY_RECORD, timestamp), node);
+        
+        updateProcessorContext(new StampedRecord(new ConsumerRecord<>(ProcessorContextImpl.NONEXIST_TOPIC, -1, -1L, null, null), timestamp), node);
```

Comment:
Why do we need to create a new record each time? Won't it be sufficient to declare `DUMMY_RECORD` as non-static?